src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -933,11 +933,7 @@ public void Sort(Comparison<T> comparison)
 
                 if (Count > 1)
                 {
-                    // Array.Sort does not have an overload that takes both bounds and a Comparison.
-                    // We could special case _count == _elements.Length in order to try to avoid
-                    // the IComparer allocation, but the Array.Sort overload that takes a Comparison
-                    // allocates such an IComparer internally, anyway.
-                    Array.Sort(_elements, 0, _count, Comparer<T>.Create(comparison));
+                    Array.Sort(_elements, comparison);
```

Comment:
```suggestion
                    _elements.AsSpan(0, _count).Sort(comparison);
```

And can you also change the `Sort` overload above without `Comparison`?

Reply:
Won't work due to array covariance I think.

Reply:
> Won't work due to array covariance I think.

Aren't all of the arrays created internally by this type as T[]? Covariance would only be an issue if _elements could be assigned to something that was U[] where U : T.

Reply:
(I think the bigger issue would be this building against a target that doesn't have MemoryExtensions.Sort.)

Reply:
Oh, just saw .NET Standard 2.0 doesn't have it.

Reply:
> > Won't work due to array covariance I think.
> 
> Aren't all of the arrays created internally by this type as T[]? Covariance would only be an issue if _elements could be assigned to something that was U[] where U : T.

Now that I look at it different typed arrays shouldn't happen here.

Reply:
I've used conditional compilation to handle the absence of `MemoryExtensions.Sort`